tools/mksurfdata_map/src/mkgridmapMod.F90

```diff
@@ -681,20 +682,19 @@ subroutine gridmap_areaave_srcmask (gridmap, src_array, dst_array, nodata, mask_
     real(r8), intent(out):: dst_array(:)
     real(r8), intent(in) :: nodata               ! value to apply where there are no input data
     real(r8), intent(in) :: mask_src(:)
+    real(r8), intent(out) :: wtnorm(:)  ! same as gridmap%frac_dst
```

Comment:
I'm thinking that a more intuitive and more efficient way to do this would be to make wtnorm an input rather than output argument. Then you can introduce a new routine whose sole purpose is to compute wtnorm - doing what's currently done in the first loop of this routine. You just need to call that routine once per source file. In contrast, the current operation recomputes wtnorm for every field on the source file; this is inefficient, and I find it confusing when an output argument gets ignored and overwritten by a following call (I am left wondering, "is it guaranteed that the output values in call 2 are the same as the output values in call 1?")